### PR TITLE
Fix #2124: Raw Watch on CustomResource does not work if name specified

### LINKED ISSUE #2124: Watch on CustomResource does not work if name specified
I'm consistently seeing following behaviour with CRDs:

```        
CustomResourceDefinitionContext crdContext = new CustomResourceDefinitionContext.Builder()
                .withGroup("crd.com")
                .withPlural("testings")
                .withScope("Namespaced")
                .withVersion("v1")
                .build();

Watch watch = new DefaultKubernetesClient().customResource(crdContext)
                .watch("default", "test1", null, null,
                new Watcher<String>() {
                    @Override
                    public void eventReceived(Action action, String resource) {
                        LOGGER.info("Event received: " + resource);
                    }
                    @Override
                    public void onClose(KubernetesClientException cause) {
                    }
                });
```

Prints nothing, not even any http client messages. If the name is dropped from watch:

```
.watch("default", null, null, null,
```

It prints:

```
[INFO] 2020-04-09 22:18:01,952 [OkHttp https://kubernetes.docker.internal:6443/...] io.fabric8.kubernetes.client.dsl.internal.RawWatchConnectionManager - Websocket opened
[INFO] 2020-04-09 22:18:01,985 [OkHttp https://kubernetes.docker.internal:6443/...] oracle.aiapps.routing.example.Testing - Event received: {"apiVersion":"crd.com/v1","kind":"TestingCrd","metadata":{"creationTimestamp":"2020-04-09T19:58:15Z","generation":4,"name":"test1","namespace":"default","resourceVersion":"2264358","selfLink":"/apis/crd.com/v1/namespaces/default/testings/test1","uid":"0925177a-7555-4d68-a714-3e423785d6d2"},"spec":{"hello":"world!"}}
```
Additional details:

CRD definition:
```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: testings.crd.com
spec:
  group: crd.com
  versions:
    - name: v1
      served: true
      storage: true
  scope: Namespaced
  names:
    plural: testings
    singular: testing
    kind: TestingCrd
    shortNames: []
  preserveUnknownFields: false
  validation:
    openAPIV3Schema:
      type: object
      properties:
        spec:
          type: object
          properties:
            hello:
              type: string
          required:
            - hello
```

Single CRD instance existing in the cluster:

```
apiVersion: crd.com/v1
kind: TestingCrd
metadata:
  name: test1
spec:
  hello: world
```

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #2124

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RawCustomResourceOperationsImpl.java [hunk 1]
```diff
@@ -696,15 +704,6 @@ public Watch watch(String namespace, String name, Map<String, String> labels, St
...
   }
 
-  private String fetchUrl(String name, String namespace, Map<String, String> labels) {
-    String url = fetchUrl(namespace, labels);
-    if (name != null) {
-      return url + name;
-    } else {
-      return url.substring(0, url.length() - 1);
-    }
-  }
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RawCustomResourceOperationsImpl.java:706]: Is it possible to keep the constructor clean and leave the additional logic here?
e.g.
```
  private String fetchUrl(String name, String namespace, Map<String, String> labels, String resourceVersion) {
    String resourceUrl = fetchUrl(namespace, labels);
    if (resourceUrl.endsWith("/")) {
      resourceUrl = resourceUrl.substring(0, resourceUrl.length() - 1);
    }
    URL url = new URL(resourceUrl);
    if (name != null) {
      httpUrlBuilder.addQueryParameter("fieldSelector", "metadata.name=" + name);
    }
    if (resourceVersion != null) {
      httpUrlBuilder.addQueryParameter("resourceVersion", resourceVersion);
    }
    return url;
  }
```

If you package expose the method, you could even add a couple of unit tests to verify the expected logic for each input variable combination.

**Comment by manusa** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**